src/app/frontend/common/components/date/component.ts

```diff
@@ -36,14 +39,41 @@ import {Component, Input} from '@angular/core';
   selector: 'kd-date',
   templateUrl: './template.html',
   styleUrls: ['./style.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class DateComponent {
+export class DateComponent implements OnInit, OnDestroy {
   @Input() date: string;
   @Input() format = 'medium';
 
   _relative: boolean;
+  changeIteration = 0;
   @Input('relative')
   set relative(v: boolean) {
     this._relative = v !== undefined && v !== false;
   }
+  private _unsubscribe = new Subject<void>();
+
+  constructor(private readonly settings_: GlobalSettingsService, private readonly cdr_: ChangeDetectorRef) {}
+
+  ngOnInit() {
```

Comment:
This feels a bit hacky. Can't we leave it on default change detection?

Reply:
The change detection does not really play the role here. The pipe for relative time is generating each time a different result as time moves on. 

However it is not set as "not pure". So even after change detection cycle you can't see the changed output text. One option is to set the `pure` flag to `false`, second option is to trigger manual update using an additional parameter that is changing over time. 

`pure: false` has a lot of disadvantages. Cause even moving your mouse over the date component triggers update/re-render. That is not really optimal and also causes change detection problems when the pipe result is changed during the change detection loop - which happens a lot if you have, let say, a new pod in the list and the text changes every second.

That's why I switched to the additional parameter that is much more optimal and safer. If it helps we can have a fixed refresh interval cause you need some trigger...

And regarding change detection strategy: `OnPush` has to be set for all sub-components to avoid unneeded change detection loop in components higher in your tree. As far as I remember form my past experiments with change detection: change detection triggered on child component marks your component for view update. So IMHO if you set `OnPush` on the list component it is not really/only partially effective until all column components are `OnPush` as well. 

And in general the speedup with `OnPush` even for a simple component like this one is huge.

Reply:
Shouldn't parent component on change event trigger also child component? Also, the date input should change over time and trigger onChange event which in result can mark component for check?

Reply:
Not in case of "pure" pipes. Such pipe is evaluated only if arguments were changed. But the date is the same string, only current time that is not an argument changes -> the pipe is not re-evaluated and text not changed, ever.

We can make current time an argument here, but that would cause an endless change detection loop. So I've chosen  a number that increments on fixed intervals instead.

Reply:
And `pure: true` is default for all pipes.

Reply:
I'd then simply define a const interval of 1 minute here to refresh the date.

Reply:
OK, will try to think about something 😉

Reply:
@floreks What about this one:
```typescript
  ngOnChanges() {
    this._unsubscribe.next();
    if (this._relative) {
      // Every 5 seconds the first minute, every minute the first hour, afterwards every hour
      of([5, 5, 11], [60, 60, 59], [3600, 3600, Infinity])
        .pipe(mergeMap(([delay, period, takeFirstN]) =>
          timer(delay * 1000, period * 1000).pipe(take(takeFirstN)))
        )
        .pipe(takeUntil(this._unsubscribe))
        .subscribe(() => {
          this.changeIteration++;
          this.cdr_.markForCheck();
        });
    }
  }
```

Reply:
Verified and tested -  https://github.com/kubernetes/dashboard/pull/5245/commits/6c5dba5c19dfa34d63af9f60e4ca3313aa8b4124